src/api/cli/integration/integration_test.go

```diff
@@ -1341,3 +1339,311 @@ func TestLiveWalletDir(t *testing.T) {
+						"0.5",
+					},
+					{
+						w.Entries[2].Address.String(),
+						"0.5",
+					},
+				}
+
+				v, err := json.Marshal(addrCoins)
+				require.NoError(t, err)
+
+				return []string{"send", "-m", string(v)}
+			},
+			nil,
+			func(t *testing.T, txid string) {
+				tx := getTransaction(t, txid)
+				// Confirms the second address receives 0.5 coin and 1 coinhour in this transaction
+				checkCoinsAndCoinhours(t, tx, w.Entries[1].Address.String(), 500000, 1)
+				// Confirms the third address receives 0.5 coin and 1 coinhour in this transaction
+				checkCoinsAndCoinhours(t, tx, w.Entries[2].Address.String(), 500000, 1)
+				// Confirms the first address has at least 1 coin left.
+				coins, _ := getAddressBalance(t, w.Entries[0].Address.String())
+				require.True(t, coins >= 1000000)
+			},
+		},
+		{
+			// Send 0.001 coin from the third address to the second address.
+			// Set the second as change address, so the 0.499 change coin will also be sent to the second address.
+			// After sending, the second address should have 1 coin and 1 coin hour.
+			"send with -c(change address) -a(from address) options",
+			func() []string {
+				return []string{"send", "-c", w.Entries[1].Address.String(),
+					"-a", w.Entries[2].Address.String(), w.Entries[1].Address.String(), "0.001"}
+			},
+			nil,
+			func(t *testing.T, txid string) {
+				tx := getTransaction(t, txid)
+				// Confirms the second address receives 0.5 coin and 0 coinhour in this transaction
+				checkCoinsAndCoinhours(t, tx, w.Entries[1].Address.String(), 500000, 0)
+				// Confirms the second address have 1 coin and 1 coin hour
+				coins, hours := getAddressBalance(t, w.Entries[1].Address.String())
+				require.Equal(t, uint64(1000000), coins)
+				require.Equal(t, uint64(1), hours)
+			},
+		},
+		{
+			// Send 1 coin from second to the the third address, this will spend three outputs(0.2, 0.3. 0.5 coin),
+			// and burn out the remaining 1 coin hour.
+			"send to burn all coin hour",
+			func() []string {
+				return []string{"send", "-a", w.Entries[1].Address.String(),
+					w.Entries[2].Address.String(), "1"}
+			},
+			nil,
+			func(t *testing.T, txid string) {
+				// Confirms that the third address has 1 coin and 0 coin hour
+				coins, hours := getAddressBalance(t, w.Entries[2].Address.String())
+				require.Equal(t, uint64(1000000), coins)
+				require.Equal(t, uint64(0), hours)
+			},
+		},
+		{
+			// Send with 0 coin hour, this test should fail.
+			"send 0 coin hour",
+			func() []string {
+				return []string{"send", "-a", w.Entries[2].Address.String(),
+					w.Entries[1].Address.String(), "1"}
+			},
+			[]byte("ERROR: Transaction has zero coinhour fee. See 'skycoin-cli send --help'"),
+			func(t *testing.T, txid string) {},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			output, err := exec.Command(binaryPath, tc.args()...).CombinedOutput()
+			if err != nil {
+				fmt.Println(string(output))
+				return
```

Comment:
In case of an error  it will just `return` and never go to `require.NoError`

Reply:
Yes, for now, the error message are in the output, if any error, we should not continue the following test code. But instead of just print the output, I should run t.Fatalf() to fail the test.

Reply:
yes, sorry that's what I meant to say that this should fail the test somewhere